src/p2p/Lost.ts

```diff
@@ -1192,8 +1192,101 @@ async function isDownCheck(node) {
   /* prettier-ignore */ if (logFlags.lost) info(`Checking internal connection for ${node.id}, cycle: ${currentCycle}`)
 
   try {
-    if (config.p2p.useProxyForDownCheck) {
-      //using the 'apoptosize' route to check if the node is up.
+    if (config.p2p.useNearNodeForDownCheck) {
+      //using the 'get_trie_hashes' route to check if the node is up.
+      let obj = { counter: currentCycle, checker: Self.id, target: node.id, timestamp: shardusGetTime() }
+      let hash = crypto.hash(obj)
+      let hashTrieReq: HashTrieReq = {
+        radixList: ['0'],
+      }
+      let closestNodes = stateManager.getClosestNodes(hash, 5, true)
+      let nearNode: P2P.NodeListTypes.Node
+      for (let closetNode of closestNodes) {
+        if (closetNode.id !== node.id) {
+          nearNode = closetNode
+          break
+        }
+      }
+      if (nearNode == null) {
+        throw new Error(`isDownCheck unable to get near node to check`)
+      }
+
+      //check radix response for near node
+      let nearNodeAsk
+      if (
+        this.stateManager.config.p2p.useBinarySerializedEndpoints &&
+        this.stateManager.config.p2p.getTrieHashesBinary
+      ) {
+        nearNodeAsk = Comms.askBinary<GetTrieHashesRequest, GetTrieHashesResponse>(
+          nearNode,
+          InternalRouteEnum.binary_get_trie_hashes,
+          hashTrieReq,
+          serializeGetTrieHashesReq,
+          deserializeGetTrieHashesResp,
+          {}
+        )
+      } else {
+        nearNodeAsk = this.p2p.ask(node, 'get_trie_hashes', hashTrieReq)
+      }
+      const nearNodeResponse = await nearNodeAsk
+
+      if (logFlags.verbose)
+        info(
+          `lost check result for nearNodeAsk near node:${node.id} cycle ${currentCycle} is ${utils.stringifyReduce(
+            nearNodeResponse
+          )}`
+        )
+
+      if (nearNodeResponse == null) {
+        /* prettier-ignore */ nestedCountersInstance.countEvent('p2p', 'isDownCheck-down-near-node-fail', 1)
+        return 'down'
```

Comment:
Does it mean "near node" is down or "target node" is down? If  we cannot get response from near node, isn't that  we report it as "lost" and try another near node?